server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/SetMessagesMethodTest.java

```diff
@@ -2327,4 +2327,198 @@ public void attachmentAndEmptyBodyShouldBeRetrievedWhenChainingSetMessagesAndGet
+            .body(ARGUMENTS + ".created", aMapWithSize(1))
+                // assert server-set attributes are returned
+            .body(ARGUMENTS + ".created", hasEntry(equalTo(messageCreationId), Matchers.allOf(
+                hasEntry(equalTo("id"), not(isEmptyOrNullString())),
+                hasEntry(equalTo("blobId"), not(isEmptyOrNullString())),
+                hasEntry(equalTo("threadId"), not(isEmptyOrNullString())),
+                hasEntry(equalTo("size"), not(isEmptyOrNullString()))
+            )))
+                // assert that message flags are all unset
+            .body(ARGUMENTS + ".created", hasEntry(equalTo(messageCreationId), Matchers.allOf(
+                hasEntry(equalTo("isDraft"), equalTo(false)),
+                hasEntry(equalTo("isUnread"), equalTo(false)),
+                hasEntry(equalTo("isFlagged"), equalTo(false)),
+                hasEntry(equalTo("isAnswered"), equalTo(false))
+            )))
+        ;
+        accessToken = JmapAuthentication.authenticateJamesUser(toUsername, password);
+        String inboxMailboxId = getMailboxId(accessToken, Role.INBOX);
+
+        calmlyAwait.atMost(60, TimeUnit.SECONDS).until( () -> messageOnInboxHasHeaders(inboxMailboxId));
+
+    }
+
+    private boolean messageOnInboxHasHeaders(String inboxMailboxId) {
+        try {
+            with()
+                .header("Authorization", accessToken.serialize())
+                .body("[[\"getMessageList\", "
+                    + "{"
+                    + "\"fetchMessages\": true, "
+                    + "\"fetchMessageProperties\": [\"headers\"], "
+                    + "\"filter\":{\"inMailboxes\":[\"" + inboxMailboxId + "\"]} "
+                    + "}, \"#0\"]]")
+                .when()
+                .post("/jmap")
+                .then()
+                .statusCode(200)
+                .body(ARGUMENTS + ".messageIds", contains("username1@domain.tld|INBOX|1"))
+                .body(SECOND_NAME, equalTo("messages"))
+                .body(SECOND_ARGUMENTS + ".list[0]", hasEntry(equalTo("headers"), Matchers.allOf(
+                    hasEntry(equalTo("Sender"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("Content-Transfer-Encoding"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("From"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("To"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("Message-ID"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("MIME-Version"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("Subject"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("Date"), not(isEmptyOrNullString())),
+                    hasEntry(equalTo("Content-Type"), not(isEmptyOrNullString()))
+                )))
+            ;
+            return true;
+        } catch(AssertionError e) {
+            e.printStackTrace();
```

Comment:
Nope, but please remove the noisy stacktrace ... If we have troubles with this test, we will add it or add logs on JMAP request.
Could you remove the printStackTrace ?